dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/HadoopMetadataConnector.java

```diff
@@ -82,13 +100,40 @@ public void addTasksTo(@Nonnull List<? super Task<?>> out, @Nonnull ConnectorArg
         .map(scriptName -> new BashTask(scriptName, HadoopScripts.extract(scriptName + ".sh")))
         .forEach(out::add);
     addTasksForSingleLineScripts(out);
+    generateServiceScripts(out);
   }
 
   private void addTasksForSingleLineScripts(List<? super Task<?>> out) {
     HadoopScripts.extractSingleLineScripts()
         .forEach((scriptName, scriptFile) -> out.add(new BashTask(scriptName, scriptFile)));
   }
 
+  private void generateServiceScripts(List<? super Task<?>> out) {
```

Comment:
I would suggest providing the finished tasks as the return value, rather than modifying the `out` parameter.

If this should have a measurable effect on performance and the copy would need to be avoided, `ImmutableList` can be replaced with a different type.

Reply:
Done.